mavros/src/plugins/local_position.cpp

```diff
@@ -91,16 +97,14 @@ class LocalPositionPlugin : public MavRosPlugin {
 				pos_ned.x, pos_ned.y, pos_ned.z,
 				pos_ned.vx, pos_ned.vy, pos_ned.vz);
 
-		/* TODO: check convertion to ENU
-		 * I think XZY is not body-fixed, but orientation does.
-		 * Perhaps this adds additional errorprone to us.
-		 * Need more tests. Issue #49.
-		 *
-		 * orientation in ENU, body-fixed
-		 */
 		tf::Transform transform;
 		transform.setOrigin(tf::Vector3(pos_ned.y, pos_ned.x, -pos_ned.z));
-		transform.setRotation(uas->get_attitude_orientation());
+		geometry_msgs::QuaternionStampedPtr q = boost::make_shared<geometry_msgs::QuaternionStamped>();
+		tf::quaternionTFToMsg(uas->get_attitude_orientation(), q->quaternion);
+		q->header.frame_id = child_frame_id;
+		q->header.stamp = uas->synchronise_stamp(pos_ned.time_boot_ms);
+		tf_listener->transformQuaternion(frame_id, *q, *q);
+		transform.setRotation(tf::Quaternion(q->quaternion.x, q->quaternion.y, q->quaternion.z, q->quaternion.w));
```

Comment:
Style: please place Q transform before `tf::Transform transform;` so filling of final transform will bi in one chunk of code.